src/entt/entity/registry.hpp

```diff
@@ -446,6 +464,35 @@ class basic_registry {
         return assure<Type>(id);
     }
 
+    /**
+     * @brief Tries to create a storage for a component with the given arguments.
+     *
+     * @warning
+     * If a storage for the given component exist already, false is returned!
+     *
+     * @tparam Type Type of component of which to return the storage.
+     * @param allocator The allocator passed to the storage.
+     * @param traits Optional component traits type passed to the storage.
+     * @param id Optional name used to map the storage within the registry.
+     * @return Returns true if the component could be created successfully.
+     *         If the component existed already the creation is skipped and false is returned.
+     */
+    template<typename Type>
```

Comment:
The reason why this function cannot be factored into the stroage is that it would be very useful to  check whether the storage was freshly created with exact the provided arguments to verify correct initialization.

Since `storage` always returns the (maybe already default constructed storage) the `storage` method is not sufficient.

The default initializatin always needs to be possible for any storage by library design.

Reply:
I might be wrong but this looks like a direct consequence of the choice of using runtime traits as a constructor argument.
If the runtime values were on the storage itself, you could create the latter and set them without the need of an extra function.